src/app.py

```diff
@@ -1,73 +1,354 @@
+        logger.warning("Defaulting page to 1")
+        page = 1
+
+    logger.debug(f"page = {page}")
+    logger.debug(f"status = {status}")
+
+    res = CatalogueItem.query.filter_by(transfer_status=status).paginate(
+        page=page,
+        per_page=CFG.ITEMS_PER_PAGE,
+        error_out=False,
+    )
+    res = CatalogueItemSchema(many=True).dump(res.items)
+    logger.debug(f"Total rows selected = {len(res)}")
+
+    return jsonify(res)
+
+
+@app.route("/catalogue/", methods=["POST"])
+@token_required
+def create_catalogue():
+    """
+    Create single catalog item record and return the created record.
+    """
+    logger.info("/catalogue/ POST called")
+    data = request.json
+
+    mandatory_fields = ["name", "checksum_algorithm", "checksum_value"]
+    for field in mandatory_fields:
+        if field not in data:
+            abort_json(
+                400,
+                error="INSERTION_FAILED",
+                message=f"Missing '{field}' in the json body...",
+            )
+    data["uuid"] = data.get("uuid", uuid.uuid4().hex)
+    data["created_on"] = data.get("created_on", datetime.now())
+    data["updated_on"] = data.get("updated_on", datetime.now())
+    data["transfer_status"] = data.get("transfer_status", "NOT_STARTED").upper()
+
+    query = CatalogueItem.query.filter_by(uuid=data["uuid"]).first()
+    if query:
+        abort_json(
+            400,
+            error="INSERTION_FAILED",
+            message=f"uuid={data['uuid']} already exists!",
+        )
+
+    try:
+        item = CatalogueItem(**data)
+        db.session.add(item)
+        db.session.commit()
+    except:
+        abort_json(
+            400,
+            error="INSERTION_FAILED",
+            message="Unable to create the catalogue item...",
+        )
+
+    return jsonify(data)
+
+
+@app.route("/catalogue/<uuid>/", methods=["PATCH"])
+@token_required
+def patch_catalogue(uuid: str):
+    """
+    Upsert a single catalogue item.
+    """
+    logger.info("/catalogue/<uuid> PATCH called")
+    data = request.json
+
+    item = CatalogueItem.query.filter_by(uuid=uuid).first()
+    if not item:
+        abort_json(400, error="PATCH_FAILED", message="uuid doesn't exist!")
+
+    try:
+
+        if "transfer_status" in data:
+            data["transfer_status"] = data["transfer_status"].upper()
+        item.update(data)
+    except:
+        abort_json(
+            400,
+            error="PATCH_FAILED",
+            message="Unable to create the catalogue item...",
+        )
 
-@app.route("/catalogue/upload/", methods=["POST"])
+    db.session.commit()
+    return jsonify(CatalogueItemSchema().dump(item))
+
+
+@app.route("/catalogue/<uuid>/", methods=["DELETE"])
+@token_required
+def delete_catalogue(uuid: str):
+    item = CatalogueItem.query.filter_by(uuid=uuid).first()
+    if not item:
+        logger.error(f"Item for uuid={uuid} not found!")
+        abort_json(400, error="DELETION_FAILED", message="uuid doesn't exist!")
```

Comment:
404, also these can be extracted into a decorator that handles finding items, assigning them to a specific value and returning status code errors if there are errors.

Reply:
I agree that we could use a better CRUD-decorator as all the endpoints are basically CRUD. But might take time. So, I am leaving it unresolved for time being till the next PR. 

cc: @ub0005